<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>producer_consumerClass</class>
 <widget class="QMainWindow" name="producer_consumerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>920</width>
    <height>505</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>producer_consumer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="selectProducerNumber">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::NoFrame</enum>
    </property>
    <property name="text">
     <string>选择生产者数量</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="producerNumberBox">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>10</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>3</number>
    </property>
    <property name="maximum">
     <number>5</number>
    </property>
   </widget>
   <widget class="QPushButton" name="confirmButton">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>10</y>
      <width>111</width>
      <height>91</height>
     </rect>
    </property>
    <property name="text">
     <string>确认</string>
    </property>
   </widget>
   <widget class="QLabel" name="selectConsumerNumber">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::NoFrame</enum>
    </property>
    <property name="text">
     <string>选择消费者数量</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="consumerNumberBox">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>40</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>3</number>
    </property>
    <property name="maximum">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="selectBufferLength">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>70</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::NoFrame</enum>
    </property>
    <property name="text">
     <string>选择存储区长度</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="bufferLengthBox">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>70</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>20</number>
    </property>
   </widget>
   <widget class="QLabel" name="tips">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>10</y>
      <width>391</width>
      <height>91</height>
     </rect>
    </property>
    <property name="text">
     <string>点击确认会根据选择的数据生成buffer的操作过程
可点击“前往”查看某一步的结果
也可以选择“下一步”、“上一步”一步步查看
表格中蓝色代表下一个插入点，红色代表下一个取出点
产物为三位，第一位为生产者编号</string>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="nextStepButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>130</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>下一步</string>
    </property>
   </widget>
   <widget class="QPushButton" name="preStepButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>130</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>上一步</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="stepBox">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>130</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="goToButton">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>130</y>
      <width>80</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>前往</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="bufferTable">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>170</y>
      <width>691</width>
      <height>61</height>
     </rect>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <property name="editTriggers">
     <set>QAbstractItemView::EditTrigger::NoEditTriggers</set>
    </property>
    <property name="cornerButtonEnabled">
     <bool>false</bool>
    </property>
    <property name="rowCount">
     <number>0</number>
    </property>
    <property name="columnCount">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLabel" name="proOrConLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>240</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="itemId">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>240</y>
      <width>54</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>920</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>生产消费</string>
    </property>
    <addaction name="separator"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="producer_consumer.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>confirmButton</sender>
   <signal>clicked()</signal>
   <receiver>producer_consumerClass</receiver>
   <slot>runMainFunction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>347</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>58</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nextStepButton</sender>
   <signal>clicked()</signal>
   <receiver>stepBox</receiver>
   <slot>stepUp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>162</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>preStepButton</sender>
   <signal>clicked()</signal>
   <receiver>stepBox</receiver>
   <slot>stepDown()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>162</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>160</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>goToButton</sender>
   <signal>clicked()</signal>
   <receiver>producer_consumerClass</receiver>
   <slot>goToStep()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>328</x>
     <y>164</y>
    </hint>
    <hint type="destinationlabel">
     <x>462</x>
     <y>143</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nextStepButton</sender>
   <signal>clicked()</signal>
   <receiver>goToButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>249</x>
     <y>174</y>
    </hint>
    <hint type="destinationlabel">
     <x>296</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>preStepButton</sender>
   <signal>clicked()</signal>
   <receiver>goToButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>68</x>
     <y>160</y>
    </hint>
    <hint type="destinationlabel">
     <x>292</x>
     <y>152</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>runMainFunction()</slot>
  <slot>goToStep()</slot>
 </slots>
</ui>
